### PR TITLE
Conditional rendering

### PR DESCRIPTION
**Pull request by neocotic**
This PR adds conditional rendering for the image element (and support for future renderers). The image element is now only rendered if it was passed to the `QRious` constructor as the `element` option OR if the `QRious#image` field is referenced at any time (even if it's not used, we can't tell the difference). The `canvas` element is still always rendered as `image` depends on it.

Not only will this help with possible Content Security Policy (CSP) issues (see #71) but I should think that it will also improve performance for those only using `canvas` elements as the browser will no longer spend time generating a data URI for the `image` element.

To summarise, the `image` element will only be rendered in the following situations:

``` javascript
const qr = new QRious({
  element: document.createElement('img'),
  value: 'https://github.com/neocotic/qrious'
});
// OR:
const qr = new QRious({
  value: 'https://github.com/neocotic/qrious'
});
qr.image;
```

### CODE REVIEW

**Comment by neocotic** [comment]
@anthonyryan1 Can you please take a look at these changes to see what you think. This should hopefully resolve #71.

**Comment by neocotic** [comment]
@anthonyryan1 Once you've completed your review, please let me know and I can get the final task for 2.3.0 completed quickly and the version released ASAP.

**Comment by neocotic** [comment]
@anthonyryan1 I'm just going to go ahead and merge this. Due to lack of tests for this project, I think the best way of checking this out is getting it released and in front of developers.

If you do happen to find anything, we can raise a bug/enhancement to get it fixed/changed in a fix release.

I don't mean to be impatient, I'm actually just enthusiastic about this project again so I won't to get it into a good state so that I can start addressing some of the more fundamental issues (e.g. support for non-ASCII characters).

**Comment by anthonyryan1** [comment]
Sorry about the delayed response, I had a chance to review this tonight and it works great!

Great work, and thank you for considering this use case. :smile:
